rosette/rbl/rosette/namespace-util.rbl

```diff
@@ -0,0 +1,119 @@
+;;; TODO: Remove unused methods in Tuplespace and add tests for the rest
+
+;;; TODO: Add tag for whether persistent
+;;; TODO: Translate match in compiler
+
+(random-number-init)
+;;; The debugSymbol is inserted so that the output is easier to read
+(defOprn generateFresh)
+(defProc (generateFresh debugSymbol) (->symbol (string-concat "Rholang-" debugSymbol "-" (->string (random-number)))))
+
+(defOprn append)
+(defPure Tuple (append item) (reverse [item & (reverse (self))]))
+
+(defOprn string-length<)
+(defOprn string-length>=)
+(defPure String (string-length< other) (< (string-length (self)) (string-length other)))
+(defPure String (string-length>= other) (not (string-length< (self) other)))
+
+(defOprn symbol-length<)
+(defOprn symbol-length>=)
+(defPure Symbol (symbol-length< other) (string-length< (->string (self)) (->string other)))
+(defPure Symbol (symbol-length>= other) (string-length>= (->string (self)) (->string other)))
+
+;;; Deletes element at position i in Tuple
+(defOprn delete-i)
+(defPure Tuple (delete-i p)
+    (concat (prim-sub-object (self) 0 p)
+        (prim-sub-object (self) (fx+ p 1)
+                 (fx- (prim-size (self)) (fx+ p 1)))))
+
+(defOprn replace-i)
+(defPure Tuple (replace-i p item)
+    (concat
+        (prim-sub-object (self) 0 p)
+        [item]
+        (prim-sub-object (self) (fx+ p 1)
+                 (fx- (prim-size (self)) (fx+ p 1)))))
+
+(defOprn symbol-list-concat)
+(defPure Tuple (symbol-list-concat) (->symbol (fold (self) (proc [e acc k] (k (string-concat acc (->string e)))) "")))
```

Comment:
Looks like this is used in as a key to a table further down. Consider whether this matters:
```
rosette> (same? (symbol-list-concat ['ab 'c]) (symbol-list-concat ['a 'bc]))
#t
```

Reply:
This could become an issue down the road. Right now its saved by the fact that all channel names are "rholang-930280" or some number so you can never have two channel symbols being equal to a third. The longer term plan would be to remove the lookup tables - I didn't want to modify the RblTable to take tuples as keys immediately.

Reply:
I'll link this conversation in a comment.

Reply:
Actually I just created an issue https://rchain.atlassian.net/browse/ROS-340